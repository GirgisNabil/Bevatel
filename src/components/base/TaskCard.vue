<script setup lang="ts">
import { Card, CardContent, CardDescription, CardHeader, CardTitle } from '@/components/ui/card'
import DeleteDialog from './DeleteDialog.vue'
import { Icon } from '@iconify/vue'
import type { ITask } from '@/types/tasks'
import EditDialog from './EditDialog.vue'

defineProps<ITask>()
</script>

<template>
  <Card class="p-4 flex justify-between">
    <CardHeader class="p-0">
      <CardTitle class="text-lg font-normal max-w-56 truncate"> {{ title }}</CardTitle>
      <CardDescription class="max-w-56 truncate">{{ description }}</CardDescription>
    </CardHeader>
    <CardContent class="p-0">
      <div class="flex items-center gap-2">
        <!-- edit dialog goes here -->
        <EditDialog :description="description" :title="title" :status="status" :id="id" />
        <DeleteDialog :id="id" />
      </div>
    </CardContent>
  </Card>
</template>
