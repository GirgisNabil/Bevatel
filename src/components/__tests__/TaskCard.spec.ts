// Import necessary libraries and types
import { mount } from '@vue/test-utils'
import EditTask from '@/components/base/EditDialog.vue'
import { describe, it, expect } from 'vitest'
import TaskCard from '@/components/base/TaskCard.vue'

// Mock sub-components used in TaskCard
const DeleteDialog = { template: '<div></div>' }
const EditDialog = { template: '<div></div>' }

// Define a test suite for TaskCard component
describe('TaskCard', () => {
  // Define a test case for rendering TaskCard
  it('renders TaskCard component correctly', async () => {
    // Mount TaskCard component
    const wrapper = mount(TaskCard, {
      props: {
        id: 1,
        title: 'Sample Task',
        description: 'This is a sample task description.',
        status: 'todo'
      },
      // Provide stubs for sub-components used in TaskCard
      global: {
        stubs: {
          DeleteDialog,
          EditDialog
        }
      }
    })

    // Assert that the TaskCard is rendered correctly
    expect(wrapper.html()).toMatchSnapshot()
  })
})
