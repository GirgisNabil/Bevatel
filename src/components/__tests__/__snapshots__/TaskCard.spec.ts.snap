// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TaskCard > renders TaskCard component correctly 1`] = `
"<div class="rounded-lg border bg-card text-card-foreground shadow-sm p-4 flex justify-between">
  <div class="flex flex-col gap-y-1.5 p-0">
    <h3 class="tracking-tight text-lg font-normal">Sample Task</h3>
    <p class="text-sm text-muted-foreground">This is a sample task description.</p>
  </div>
  <div class="p-0">
    <div class="flex items-center gap-2">
      <!-- edit dialog goes here -->
      <div description="This is a sample task description." title="Sample Task" status="todo" id="1"></div>
      <div id="1"></div>
    </div>
  </div>
</div>"
`;
